controllers/complianceeventsapi/server.go

```diff
@@ -671,6 +702,20 @@ OFFSET %d ROWS;`,
 			return
 		}
 
+		isAuth, err := getReadAuth(config, r.Header.Get("Authorization"), ce.Cluster.Name)
```

Comment:
This isn't quite right. On the "list" API endpoint, you'll need to find the clusters the user has access to and modify the `where` clause so that it's like `WHERE clusters.name  IN ('cluster1', 'cluster2', ...)` unless the user has `*` access in `resourceNames` for the `ManagedCluster` objects, then there is no filter. You'll also need to validate any `cluster.name` filters.

The thing I'm unsure about is how to handle `cluster.cluster_id` filters. I'm guessing for now, we just don't allow filtering by `cluster_id` unless the user has `*` access as mentioned above.

Reply:
Oh oh I see!!

Reply:
For example, you'd call this:
```
curl -k -s -XPOST --data-binary '{"kind": "SelfSubjectRulesReview", "apiVersion": "authorization.k8s.io/v1", "spec": {"namespace": "does-not-exist-but-is-requred"}}'  -H "Content-Type: application/json" -H "Accept: application/json" -H "Authorization: Bearer $(oc whoami --show-token)" "$(oc whoami --show-server)/apis/authorization.k8s.io/v1/selfsubjectrulesreviews" | jq .
```

Then you would do similar filters to https://github.com/stolostron/rbac-api-utils/blob/159deac7d398c3470ae730c3fbb8ffbbcca1ac5a/pkg/rbac/rbac.go#L187.